symengine/matrix.h

```diff
@@ -172,12 +172,12 @@ class DenseMatrix : public MatrixBase
                                 unsigned row_start, unsigned col_start,
                                 unsigned row_end, unsigned col_end,
                                 unsigned row_step, unsigned col_step);
-    friend void row_join(const DenseMatrix &A, const DenseMatrix &B,
-                         DenseMatrix &C);
-    friend void col_join(const DenseMatrix &A, const DenseMatrix &B,
-                         DenseMatrix &C);
-    friend void row_del(DenseMatrix &A, unsigned k);
-    friend void col_del(DenseMatrix &A, unsigned k);
+    virtual void row_join(const DenseMatrix &B);
+    virtual void col_join(const DenseMatrix &B);
+    virtual void row_insert(const DenseMatrix &B, unsigned pos);
+    virtual void col_insert(const DenseMatrix &B, unsigned pos);
+    virtual void row_del(unsigned k);
+    virtual void col_del(unsigned k);
```

Comment:
Since other functions such as `add_matrix` is also `virtual`, I thought these should also follow suit.